how-to-ns/src/com/gfredericks/how_to_ns.clj

```diff
@@ -99,24 +99,41 @@
       (apply vector ns-sym
              (apply concat (sort map-args))))))
 
+(defn maybe-normalize-require [require-clause opts]
+  (if (reader-conditional? require-clause)
+    [require-clause]
+    (normalize-require require-clause opts)))
+
 (defn normalize-imports
   [import-clauses opts]
   (let [all-classes (mapcat (fn [import-clause]
-                              (if (symbol? import-clause)
+                              (cond
+                                (symbol? import-clause)
+                                [import-clause]
+
+                                (reader-conditional? import-clause)
                                 [import-clause]
+                                
+                                :else
                                 (for [sym (rest import-clause)]
                                   (symbol (str (first import-clause) \. sym)))))
                             import-clauses)]
     (->> all-classes
-         (map split-symbol)
+         (map (fn [x]
+                (if (reader-conditional? x)
+                    [x]
+                    (split-symbol x))))
          (group-by first)
-         (sort)
+         (sort-by (fn [[x]]
+                    (str x)))
```

Comment:
`x` might be a reader conditional here, is that correct?

according to my repl, `str` on a reader conditional gives something like `clojure.lang.ReaderConditional@5044f1cd`, which seems like it would yield a nondeterministic sort here; `pr-str` would likely do better; I'd like something that involves the contents of the conditional, but I don't know the best way to do that; if you know what order the kv-pairs inside it would come in, you could sort based on the first value.

```

Reply:
> x might be a reader conditional here, is that correct?

I scienced and yes, correct. Will be fixing